clang/lib/Basic/Targets/AMDGPU.cpp

```diff
@@ -309,7 +309,6 @@ void AMDGPUTargetInfo::getTargetDefines(const LangOptions &Opts,
                         Twine("\"") + Twine(CanonName) + Twine("\""));
     Builder.defineMacro("__amdgcn_target_id__",
                         Twine("\"") + Twine(*getTargetID()) + Twine("\""));
-    Builder.defineMacro("__CUDA_ARCH__", "0");
```

Comment:
NOTE:
This was previously added to avoid setting it to a non-zero value but is not needed anymore after the changes in this PR completely disable the definition of the macro for SYCL offload. Additionally, the upstream llvm-project doesn't have  this line https://github.com/llvm/llvm-project/blob/main/clang/lib/Basic/Targets/AMDGPU.cpp#L311.